app/Http/Controllers/ActivityController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Exception;
+use Illuminate\Http\Request;
+use App\Models\ActivityUser;
+use App\Http\Request\Validate;
+use App\Facades\Util;
+use App\Helpers\ResponseInterface;
+use App\Helpers\ResourceInterface;
+use App\Http\Request\ValidationRules;
+use App\Http\Request\ValidationMessages;
+use App\Http\Resources\ErrorResource;
+
+class ActivityController extends Controller
+{
+    public function __construct(ResponseInterface $respProvider, ResourceInterface $resProvider, ValidationRules $rules, ValidationMessages $validationMessages)
+    {
+        $this->rules = $rules;
+        $this->validationMessages = $validationMessages;
+        $this->resourceProvider  = $resProvider;
+        $this->resProvider = $respProvider;
+    }
+
+    public function getActivityLog(Request $request, Validate $validate)
+    {
+        $validationErrors = $validate->validate($request, $this->rules->getActivityLogValidationRules(), $this->validationMessages->getActivityLogValidationMessages());
+        if ($validationErrors) {
+            return (new ErrorResource($validationErrors))->response()->setStatusCode(400);
+        }
+        $perPage = !empty($request->per_page) ? $request->per_page : config('global.per_page');
```

Comment:
I think it can be replaced as 
`$perPage = $request->per_page ??  : config('global.per_page');`